heartbeat/monitors/wrappers/monitors_test.go

```diff
@@ -316,3 +323,44 @@ func summaryValidator(up int, down int) validator.Validator {
 		},
 	})
 }
+
+func TestTimespan(t *testing.T) {
+	now := time.Now()
+	sched10s, err := schedule.Parse("@every 10s")
+	require.NoError(t, err)
+
+	type args struct {
+		started time.Time
+		sched   *schedule.Schedule
+		timeout time.Duration
+	}
+	tests := []struct {
+		name string
+		args args
+		want common.MapStr
+	}{
+		{
+			"interval longer than timeout",
+			args{now, sched10s, time.Second},
+			common.MapStr{
+				"gte": now,
+				"lt":  now.Add(time.Second * 10),
+			},
+		},
+		{
+			"timeout longer than interval",
```

Comment:
I keep looking at this and thinking: Shouldn't it be 10s that is reported in this case as it is the maximum we need to look at? Or it is because interval is longer it could be that multiple calls overlap?

Reply:
this is here to account  for the check being delayed for up to 20s, firing, then having a request take another 10s before timing out.